src/utils/LaserScanner.lua

```diff
@@ -64,14 +64,22 @@ function LaserScanner:getLaserData(laser_scan_array, x, y, z, dx_r, dy, dz_r)
     -- if laser_scan.ignore_terrain is set true then ignore the terrain when detected
 
     if self.vehicle_table.laser_scan.ignore_terrain then
-        if self.raycastDistance ~= self.INIT_RAY_DISTANCE and self.raycastTransformId ~= g_currentMission.terrainRootNode and self.raycastTransformId ~= self.vehicle.components[1].node then
-            table.insert(laser_scan_array, self.raycastDistance)
+        -- some vehicles do not have a 2nd component
+        if not self.vehicle.components[2] then
+            if self.raycastDistance ~= self.INIT_RAY_DISTANCE and self.raycastTransformId ~= g_currentMission.terrainRootNode and self.raycastTransformId ~= self.vehicle.components[1].node then
+                table.insert(laser_scan_array, self.raycastDistance)
+            else
+                table.insert(laser_scan_array, self.INIT_RAY_DISTANCE)
+            end
         else
-            table.insert(laser_scan_array, self.INIT_RAY_DISTANCE)
+            if self.raycastDistance ~= self.INIT_RAY_DISTANCE and self.raycastTransformId ~= g_currentMission.terrainRootNode and self.raycastTransformId ~= self.vehicle.components[1].node and  self.raycastTransformId ~= self.vehicle.components[2].node  then
+                table.insert(laser_scan_array, self.raycastDistance)
+            else
+                table.insert(laser_scan_array, self.INIT_RAY_DISTANCE)
+            end
```

Comment:
Could you perhaps add some comments which explain what's happening here?

Why do we 'care' about whether or not there is a "second component" or not?

Reply:
Added. It should be clearer now.